projects/plugins/protect/src/class-protect-status.php

```diff
@@ -0,0 +1,222 @@
+<?php
+/**
+ * Class to handle the Protect Status of Jetpack Protect
+ *
+ * @package automattic/jetpack-protect-plugin
+ */
+
+namespace Automattic\Jetpack\Protect;
+
+use Automattic\Jetpack\Connection\Client;
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+use Automattic\Jetpack\Plugins_Installer;
+use Automattic\Jetpack\Redirect;
+use Automattic\Jetpack\Sync\Functions as Sync_Functions;
+use Jetpack_Options;
+use WP_Error;
+
+/**
+ * Class that handles fetching and caching the Status of vulnerabilities check from the WPCOM servers
+ */
+class Protect_Status extends Status {
+
+	/**
+	 * WPCOM endpoint
+	 *
+	 * @var string
+	 */
+	const REST_API_BASE = '/sites/%d/jetpack-protect-status';
+
+	/**
+	 * Name of the option where status is stored
+	 *
+	 * @var string
+	 */
+	protected static $option_name = 'jetpack_protect_status';
+
+	/**
+	 * Name of the option where the timestamp of the status is stored
+	 *
+	 * @var string
+	 */
+	protected static $option_timestamp_name = 'jetpack_protect_status_time';
+
+	/**
+	 * Gets the current status of the Jetpack Protect checks
+	 *
+	 * @return Status_Model
+	 */
+	public static function get_status() {
+		if ( self::$status !== null ) {
+			return self::$status;
+		}
+
+		if ( ! self::should_use_cache() || self::is_cache_expired() ) {
```

Comment:
I'm wondering if it makes sense for us to cache stuff from the Scan API, we allow users to initiate a new scan through Calypso so we may end up serving old data in those instances. 

Perhaps it might be a good idea for us to add a task so the scan backend can invalidate the cache, we can probably add a new transport server operation to do that and call it when the scan finishes.

Reply:
I've [added an endpoint](https://github.com/Automattic/jetpack/pull/26119/commits/6cbfe0a9bd3c533b0387ca791da5aa6c85121a30) to externally trigger the resetting of the Scan cache, and also [reduced the cache time](https://github.com/Automattic/jetpack/pull/26119/commits/82d7686a0bf4d692fdb61073989f88889c180a7e) down to 5 minutes for the Scan API data source.

Based on discussion here: p1663724670556269-slack-C029WFNV69M